code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -130,6 +142,28 @@
 		to_chat(usr, "<span class='notice'>You empty \the [src] onto the floor.</span>")
 		reagents.reaction(usr.loc)
 		src.reagents.clear_reagents()
+		T = get_turf(src)
+		area = get_area(src)
+		if(reagents.has_reagent("sacid"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped sulphuric acid from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+			log_game("[key_name(user)] dumped sulphuric acid from \a [src] at [area] ([T.x], [T.y], [T.z]).")
+		if(reagents.has_reagent("facid"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped Fluacid from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+			log_game("[key_name(user)] dumped Fluacid from \a [src] at [area] [COORD(T)].")
+		if(reagents.has_reagent("lube"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped Space lube from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+			log_game("[key_name(user)] dumped Space lube from \a [src] at [area] [COORD(T)].")
+		if(reagents.has_reagent("clf3"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped CLF3 from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+			log_game("[key_name(user)] dumped CLF3 from \a [src] at [area] [COORD(T)].")
+		if(reagents.has_reagent("phlogiston"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped phlogiston from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+			log_game("[key_name(user)] dumped phlogiston from \a [src] at [area] [COORD(T)].")
+		if(reagents.has_reagent("plasma"))
+			message_admins("[ADMIN_LOOKUPFLW(user)] dumped dumped from \a [src] at [area] [ADMIN_COORDJMP(T)].")

```

Comment:
`dumped dumped`. Make the list of reagents a list, it avoids mistakes you make with copying lke this.